test/components/error_spec.js

```diff
@@ -0,0 +1,40 @@
+import PlayerError from '../../src/components/error'
+import Core from '../../src/components/core'
+import Events from '../../src/base/events'
+
+describe('PlayerError', function() {
+  beforeEach(function() {
+    this.core = new Core({})
+
+    this.errorData = {
+      code: 'test_01',
+      description: 'test error',
+      level: PlayerError.Levels.FATAL,
+      origin: 'test',
+      scope: 'it',
+      raw: {},
+    }
+  })
+
+  it('trigger ERROR with error data when error method is called', function() {
+    sinon.spy(this.core, 'trigger')
+    PlayerError.error(this.errorData)
+
+    assert.ok(this.core.trigger.calledWith(Events.ERROR, {
+      code: 'test_01',
+      description: 'test error',
+      level: PlayerError.Levels.FATAL,
+      origin: 'test',
+      scope: 'it',
+      raw: {},
+    }))
+  })
+
+  it('does not trigger ERROR when core is not setted', function() {
```

Comment:
Here I also think we can use context:

```
describe("when core is not ready", ....)

Reply:
improved on e6917a0eeb6bacfe6985cefc5a147bbcc12681cb and 8404cf9a80e2273e3879e426597d5f70a1960294